components/eam/src/physics/cam/nucleate_ice_cam.F90

```diff
@@ -503,29 +494,22 @@ subroutine nucleate_ice_cam_calc( &
 
             if (dmc > 0._r8) then
                wght = dmc/(ssmc + dmc + so4mc + bcmc + pommc + soamc + mommc)
-               dst3_num = wght * num_coarse(i,k)*rho(i,k)*1.0e-6_r8
-            else 
-               dst3_num = 0.0_r8
+               dst3_num = wght * num_coarse(i,k)*rho(i,k)*num_m3_to_cm3
             endif
 
             dst_num = dst3_num
 
             if (dgnum(i,k,mode_aitken_idx) > 0._r8) then
                ! only allow so4 with D>0.1 um in ice nucleation
-               so4_num  = num_aitken(i,k)*rho(i,k)*1.0e-6_r8 &
+               so4_num  = num_aitken(i,k)*rho(i,k)*num_m3_to_cm3 &
                           * (0.5_r8 - 0.5_r8*erf(log(so4_sz_thresh_icenuc/dgnum(i,k,mode_aitken_idx))/  &
-                          (2._r8**0.5_r8*log(sigmag_aitken))))
-            else 
-               so4_num = 0.0_r8 
+                          (2._r8**0.5_r8*alnsg_amode(modeptr_aitken))))
             endif
             so4_num = max(0.0_r8, so4_num)
 
-            ! *** Turn off soot nucleation ***
-            soot_num = 0.0_r8
-
             call nucleati( &
-               wsubi(i,k), t(i,k), pmid(i,k), relhum(i,k), icldm(i,k),   &
-               rho(i,k), so4_num, dst_num, soot_num,                     &
+               wsubi(i,k), temperature(i,k), pmid(i,k), relhum(i,k), icldm(i,k),   &
```

Comment:
maybe add in/out information for the subroutine?

Reply:
added comments